src/frameworks/next/index.ts

```diff
@@ -126,6 +127,18 @@ export async function build(
 
   const env = { ...process.env };
 
+  // Check if the .env.<PROJECT-ID> file exists and make it available for the build process
+  if (context?.projectId) {
+    const projectEnvPath = join(dir, `.env.${context.projectId}`);
+
+    if (await pathExists(projectEnvPath)) {
+      const projectEnvVars = parseStrict((await readFile(projectEnvPath)).toString());
+
+      // Merge the parsed variables with the existing environment variables
+      Object.assign(env, projectEnvVars);
```

Comment:
dotenv does not override environment variables by default if i recall, so this logic will likely need inverting `{ ...projectEnvVars, ...process.env }`

Reply:
@jamesdaniels I just made this change, added tests for the `parseStrict` function and added the changelog.